src/ToolBox/superpmi/superpmi/neardiffer.cpp

```diff
@@ -67,14 +67,51 @@ bool NearDiffer::InitAsmDiff()
 
     if (UseCoreDisTools)
     {
-        const char* coreDisToolsLibrary = MAKEDLLNAME_A("coredistools");
+#ifdef PLATFORM_UNIX
+        // Unix will require the full path to coredistools. Assume that the
+        // location is next to the full path to the superpmi.so.
+        
+        wchar_t coreCLRLoadedPath[MAX_LONGPATH];
+        HMODULE result = 0;
+        int returnVal = ::GetModuleFileNameW(result, coreCLRLoadedPath, MAX_LONGPATH);
+
+        if (returnVal == 0)
+        {
+            LogError("GetModuleFileNameW failed (0x%08x)", ::GetLastError());
+            return false;
+        }
+
+        wchar_t* ptr;
+        for (ptr = coreCLRLoadedPath + ::wcsnlen(coreCLRLoadedPath, MAX_LONGPATH) - 1; ptr != coreCLRLoadedPath; --ptr)
+        {
+            if (*ptr == '/')
+            {
+                ++ptr;
+                break;
+            }
+        }
+
+        const wchar_t* coreDisToolsLibrary = MAKEDLLNAME_W("coredistools");
+        ::wcscpy_s(ptr, ::wcsnlen(coreDisToolsLibrary, MAX_LONGPATH) + 1, coreDisToolsLibrary);
```

Comment:
Isn't the second arg to wcscpy_s the number of characters available in the destination, which would be `&coreCLRLoadedPath[MAX_LONGPATH] - ptr`?